README.md
# Schedule Selected Content

**INACTIVE NOTICE: This plugin is unsupported by WPMUDEV, we've published it here for those technical types who might want to fork and maintain it for their needs.**

## Translations

Translation files can be found at https://github.com/wpmudev/translations

## Schedule Selected Content allows you to schedule any content, in any post, to show at a time you set in the future.

And it's brilliantly easy to use. Choose to make certain post or page content available only at scheduled periods including:

*   Onetime - Choose a date in the future to start showing the content
*   Weekly - Show your content only on certain days of the week at a given time
*   Monthly - Schedule your content to display on the same day each month

There's even an awesome countdown timer that reloads the page at the beginning and end of the scheduled period to help enforce the protection. Simply install and activate using the [WPMU DEV dashboard](http://premium.wpmudev.org/project/wpmu-dev-dashboard/) and a nice new 'Schedule' icon will be added to your Visual editor that you can use to schedule selected content. 

![scheduled-option](http://premium.wpmudev.org/wp-content/uploads/2011/03/scheduled-option.png)


 Find a new icon in the content editor

 Simply select what sections of your content you want to schedule, when it's to be released and how long the content should be available. 

![schedule-selected-content](http://premium.wpmudev.org/wp-content/uploads/2011/03/schedule-selected-content1.png)


 Enter anything like to display prior to the content being released

 Easily use it for videos, audio, competitions or any other kind of media you like. Now your users, readers or students will only be able to see what you want to display at the scheduled time. 

![Example of schedule content release](http://premium.wpmudev.org/wp-content/uploads/2011/03/schedule65.jpg)

 See it in action

 Easy, effective and pretty darn powerful too. Schedule Selected Content for time controlled content sharing.


## Usage

### To Get Started:

Start by reading [Installing plugins](https://premium.wpmudev.org/wpmu-manual/installing-regular-plugins-on-wpmu/) section in our comprehensive [WordPress and WordPress Multisite Manual](https://premium.wpmudev.org/wpmu-manual/) if you are new to WordPress.

Login to your admin panel for WordPress or Multisite and activate the plugin:

*   On regular WordPress installs – visit Plugins and Activate the plugin.
*   For WordPress Multisite installs – Activate it blog-by-blog (say if you wanted to make it a Pro Sites premium plugin), or visit Network Admin -> Plugins andNetwork Activate the plugin.

### To Use:

Once activated you'll see a nice new '_Schedule_' clock icon added to your Visual editor that you can use to schedule selected content. 

![Schedule Selected Content - Schedule icon](https://premium.wpmudev.org/wp-content/uploads/2011/03/Schedule-Selected-Content-Schedule-icon.png)

 Simply select the content you'd like to schedule. 

![Schedule Selected Content - select the content](https://premium.wpmudev.org/wp-content/uploads/2011/03/Schedule-Selected-Content-select-the-content.png)

 Then click the '_Schedule_' icon, this will display a pop-up dialog where you can configure the settings. 

![Schedule Selected Content - Schedule pop-up](https://premium.wpmudev.org/wp-content/uploads/2011/03/Schedule-Selected-Content-Schedule-pop-up.png)

 Select a frequency from the _Schedule_ drop-down which will show different settings based on your selection from the following:

*   _Monthly_ - you'll be able to select a day of the month.
*   _Weekly_ - you can tick the days of the week you want the content to show.
*   _Onetime_ - you'll be able to select the exact date you want the content to be displayed on.

Specify the time you want the content to appear using the _Open Time_ settings. Tick the '_Use WordPress Time Instead of GMT_' option to use the WordPress time zone you specified at _Settings > General_. Choose how long you want the content to display using the _Open Length_ settings. Optionally, you can enter a _Message_ to be displayed before the content is made available. Click the _Insert_ button to insert the plugin's shortcode. You'll see it inserts a shortcode around your selected content. 

![Schedule Selected Content - Schedule shortcode](https://premium.wpmudev.org/wp-content/uploads/2011/03/Schedule-Selected-Content-Schedule-shortcode.png)

 On the front end you'll see a countdown clock indicating when the content will be released. 

![Schedule Selected Content - Front-end](https://premium.wpmudev.org/wp-content/uploads/2011/03/Schedule-Selected-Content-Front-end.png)


changelog.txt
Plugin Name: Scheduled Content
Author: Aaron Edwards (incsub)

Change Log:
----------------------------------------------------------------------
----------------------------------------------------------------------

1.2.2 - 8/31/2017
----------------------------------------------------------------------
- Fixed shortcode button in WP editor
- Fixed PHP warning shown in WP 4.8

1.2.1 - 9/20/2016
----------------------------------------------------------------------
- PHP7 compatibility fixes
- Update dash notice

1.2 - 6/08/2014
----------------------------------------------------------------------
- Add support for using local WP time instead of GMT

1.1.2 - 5/22/2013
----------------------------------------------------------------------
- Fix weekly schedules with only Sunday selected

1.0
----------------------------------------------------------------------
- Initial Release

includes/editor_plugin.js
(function() {
	// Load plugin specific language pack
	//tinymce.PluginManager.requireLangPack('protect');

	tinymce.create('tinymce.plugins.ScheduledContent', {
		/**
		 * Initializes the plugin, this will be executed after the plugin has been created.
		 * This call is done before the editor instance has finished it's initialization so use the onInit event
		 * of the editor instance to intercept that event.
		 *
		 * @param {tinymce.Editor} ed Editor instance that the plugin is initialized in.
		 * @param {string} url Absolute URL to where the plugin is located.
		 */
		init : function(ed, url) {
			// Register the command so that it can be invoked by using tinyMCE.activeEditor.execCommand('mceChat');
			ed.addCommand('mceScheduled', function() {
				ed.windowManager.open({
					file : ajaxurl + "?action=scheduledTinymceOptions",
    	 		width : 500,
					height : 310,
					inline : 1
				}, {
					plugin_url : url // Plugin absolute URL
				});
			});

			// Register chat button
			ed.addButton('scheduled', {
				title : "Schedule",
				cmd : 'mceScheduled',
				image : url + '/clock-icon.png'
			});

			// Add a node change handler, selects the button in the UI when a image is selected
			ed.onNodeChange.add(function(ed, cm, n) {
				cm.setActive('scheduled', n.nodeName == 'IMG');
			});
		},

		/**
		 * Creates control instances based in the incomming name. This method is normally not
		 * needed since the addButton method of the tinymce.Editor class is a more easy way of adding buttons
		 * but you sometimes need to create more complex controls like listboxes, split buttons etc then this
		 * method can be used to create those.
		 *
		 * @param {String} n Name of the control to create.
		 * @param {tinymce.ControlManager} cm Control manager to use inorder to create new control.
		 * @return {tinymce.ui.Control} New control instance or null if no control was created.
		 */
		createControl : function(n, cm) {
			return null;
		},

		/**
		 * Returns information about the plugin as a name/value array.
		 * The current keys are longname, author, authorurl, infourl and version.
		 *
		 * @return {Object} Name/value array containing information about the plugin.
		 */
		getInfo : function() {
			return {
				longname : 'Scheduled Content Plugin',
				author : 'Aaron Edwards',
				authorurl : 'http://premium.wpmudev.org',
				infourl : 'http://premium.wpmudev.org/project/scheduled-content',
				version : "1.0"
			};
		}
	});

	// Register plugin
	tinymce.PluginManager.add('scheduled', tinymce.plugins.ScheduledContent);
})();


scheduled-content.php
<?php
/*
Plugin Name: Scheduled Content
Description: Allows you to make certain post or page content available only at scheduled periods via a simple shortcode.
Plugin URI: https://premium.wpmudev.org/project/scheduled-content/
Version: 1.2.2
Author: WPMU DEV
Author URI: https://premium.wpmudev.org/
WDP ID: 215
Textdomain: sc
*/

/*
Copyright 2007-2016 Incsub (http://incsub.com)

Author: Aaron Edwards
Contributors: Dharmendra Vekariya

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License (Version 2 - GPLv2) as published by
the Free Software Foundation.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/


class ScheduledContent {

	function __construct() {

		//shortcodes
		add_shortcode( 'scheduled', array( &$this, 'shortcode' ) );

		//localize the plugin
		add_action( 'init', array( &$this, 'localization' ) );

		// TinyMCE options
		add_action( 'wp_ajax_scheduledTinymceOptions', array( &$this, 'tinymce_options' ) );
		add_action( 'admin_init', array( &$this, 'load_tinymce' ) );

		global $wpmudev_notices;
		$wpmudev_notices[] = array(
			'id'      => 215,
			'name'    => 'Scheduled Content',
			'screens' => array(
				'post'
			)
		);
		include_once( dirname( __FILE__ ) . '/includes/dash-notice/wpmudev-dash-notification.php' );
	}

	function localization() {
		// Load up the localization file if we're using WordPress in a different language
		// Place it in this plugin's "languages" folder and name it "sc-[value in wp-config].mo"
		load_plugin_textdomain( 'sc', false, dirname(plugin_basename(__FILE__)) . '/languages/' );
	}

	function shortcode( $atts, $content = null ) {
		extract( shortcode_atts( array(
			'monthly' => false, //day of month
			'weekly'  => false, //week days comma separated, 0-6 = Sunday-Saturday
			'onetime' => false, //date string, ie "08/15/2010"
			'time'    => false, //time string, ie "8:00 PM"
			'length'  => false, //how long to keep open in years:days:hours:minutes, ie "0:0:12:35"
			'msg'     => false,
			'wptime'  => false
		), $atts ) );

		//skip check for no content
		if ( is_null( $content ) ) {
			return;
		}

		//if required fields not set don't protect
		if ( ( ! $monthly && false === $weekly && ! $onetime ) || ! $time || ! $length ) {
			return do_shortcode( $content );
		}

		if ( $wptime == true ) {
			$now = current_time( 'timestamp' );
		} else {
			$now = time();
		}
		//calculate how long to keep open
		@list( $yrs, $dys, $hrs, $mns ) = explode( ':', $length );

		$length = strtotime( "+$yrs Years $dys Days $hrs Hours $mns Minutes", $now ) - $now;

		//do our checks for if its open
		$open = false;
		if ( $monthly ) {
			$start = strtotime( date( 'F' ) . ' ' . $monthly . ' ' . $time, $now );
			$end   = $start + $length;
			if ( $start <= $now && $end >= $now ) {
				$open = true;
			} else if ( $end < $now ) {
				$start = strtotime( date( 'F', strtotime( '+1 month', $now ) ) . ' ' . $monthly . ' ' . $time, $now );
				$end   = $start + $length;
				if ( $start <= $now && $end >= $now ) {
					$open = true;
				} else {
					$open = false;
				}
			}
		} else if ( false !== $weekly ) {
			$weekdays = explode( ',', $weekly );
			sort( $weekdays );

			$days = array( 0 => 'sun', 1 => 'mon', 2 => 'tue', 3 => 'wed', 4 => 'thu', 5 => 'fri', 6 => 'sat' );
			foreach ( $weekdays as $day ) {
				$periods[ strtotime( "last $days[$day] $time", $now ) ] = strtotime( "last $days[$day] $time", $now ) + $length;
				$periods[ strtotime( "this $days[$day] $time", $now ) ] = strtotime( "this $days[$day] $time", $now ) + $length;
				$periods[ strtotime( "next $days[$day] $time", $now ) ] = strtotime( "next $days[$day] $time", $now ) + $length;
			}

			ksort( $periods );
			//see if we are in one of the periods
			foreach ( $periods as $start => $end ) {
				if ( $start <= $now && $end >= $now ) {
					$open = true;
					break;
				}
			}

			//not in a period, figure out the next one
			if ( ! $open ) {
				foreach ( $periods as $start => $end ) {
					if ( $start > $now ) {
						break;
					}
				}
			}

		} else if ( $onetime ) {
			$start = strtotime( "$onetime $time", $now );
			$end   = $start + $length;
			if ( $start <= $now && $end >= $now ) {
				$open = true;
			}
		}

		//set default closed messages
		if ( ! $msg ) {
			$msg = ( $onetime && $end <= $now ) ? __( "This content is not currently available.", 'sc' ) : __( "This content is not currently available. It will be available in:", 'sc' );
		}

		//check cookie for password
		if ( $open ) {

			//refresh page at end of period
			if ( $end >= $now ) {
				$refresh = ( $end - $now ) * 1000;
				if ( $refresh < 172800000 ) //prevent integer overflow http://stackoverflow.com/questions/3468607/why-does-settimeout-break-for-large-millisecond-delay-values
				{
					$content .= '<script language="javascript">setTimeout("location.reload(true)", ' . $refresh . ');</script>';
				}
			}

			return do_shortcode( $content );
		} else {
			$return    = '<p class="scheduled-closed">' . $msg . '</p>';
			$wpgmt     = ( $wptime == true ) ? "" : " GMT";
			$var       = 'cd_' . rand();
			$countdown = '<script language="javascript" src="' . plugins_url(dirname(plugin_basename(__FILE__)) . '/includes/countdown.js') . '"></script>
<div class="scheduled-timer" id="clock_' . $var . '"></div>
<script language="javascript">
	var ' . $var . ' = new countdown("' . $var . '");
	' . $var . '.Div			= "clock_' . $var . '";
	' . $var . '.TargetDate		= "' . date( "m/d/Y g:i A", $start ) . $wpgmt . '";
	' . $var . '.DisplayFormat	= "' . __( "%%D%% Days, %%H%% Hours, %%M%% Minutes, %%S%% Seconds", 'sc' ) . '";
	' . $var . '.Setup();
</script>
			';

			//show countdown if start is in the future
			if ( $start > $now ) {
				$return .= $countdown;
			}

			return $return;
		}
	}

	function load_tinymce() {
		if ( ( current_user_can( 'edit_posts' ) || current_user_can( 'edit_pages' ) ) && get_user_option( 'rich_editing' ) == 'true' ) {
			add_filter( 'mce_external_plugins', array( &$this, 'tinymce_add_plugin' ) );
			add_filter( 'mce_buttons', array( &$this, 'tinymce_register_button' ) );
		}
	}

	/**
	 * TinyMCE dialog content
	 */
	function tinymce_options() {
		?>
		<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
		<html>
		<head>
			<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
			<?php
				/**
				 * TODO: We are including our own copy of tiny_mce_popup.js until this issue is resolved in the core file: https://core.trac.wordpress.org/ticket/41124
				 */
			?>
            <script type="text/javascript"
                    src="<?php echo plugins_url(dirname(plugin_basename(__FILE__)) . '/includes/tiny_mce_popup.js'); ?>">
            </script>
			<script type="text/javascript" src="../wp-includes/js/tinymce/utils/form_utils.js?ver=327-1235"></script>
			<script type="text/javascript" src="../wp-includes/js/tinymce/utils/editable_selects.js?ver=327-1235"></script>

			<script type="text/javascript" src="../wp-includes/js/jquery/jquery.js"></script>

			<script type="text/javascript">

				tinyMCEPopup.storeSelection();

				jQuery(document).ready(function ($) {
					$('#sc-type').change(function () {
						if ($(this).val() == 'monthly') {
							$('#sc-weekly-row').hide();
							$('#sc-onetime-row').hide();
							$('#sc-monthly-row').show();
						} else if ($(this).val() == 'weekly') {
							$('#sc-monthly-row').hide();
							$('#sc-onetime-row').hide();
							$('#sc-weekly-row').show();
						} else if ($(this).val() == 'onetime') {
							$('#sc-monthly-row').hide();
							$('#sc-weekly-row').hide();
							$('#sc-onetime-row').show();
						}
					});
				});


				var insertSchedule = function (ed) {

					tinyMCEPopup.restoreSelection();

					output = '[scheduled';

					//insert schedule type
					if (jQuery('#sc-type').val() == 'monthly') {
						output += ' monthly="' + jQuery('#sc-monthdate').val() + '"';
					} else if (jQuery('#sc-type').val() == 'weekly') {
						var weekdays = '';
						jQuery('.sc-weekdays:checked').each(function (index) {
							weekdays = weekdays + ',' + jQuery(this).val();
						});
						if (!weekdays) {
							alert("<?php _e("You must select at least one weekday.", 'sc'); ?>");
							return false;
						}
						output += ' weekly="' + weekdays.substring(1) + '"';
					} else if (jQuery('#sc-type').val() == 'onetime') {
						output += ' onetime="' + jQuery('#sc-year').val() + '/' + jQuery('#sc-month').val() + '/' + jQuery('#sc-day').val() + '"';
					}

					//insert start time
					output += ' time="' + jQuery('#sc-hours').val() + ':' + jQuery('#sc-minutes').val() + ' ' + jQuery('#sc-ampm').val() + '"';

					//insert open length
					output += ' length="' + jQuery('#sc-lyears').val() + ':' + jQuery('#sc-ldays').val() + ':' + jQuery('#sc-lhours').val() + ':' + jQuery('#sc-lmins').val() + '"';

					//insert msg
					if (jQuery('#sc-msg').val())
						output += ' msg="' + jQuery('#sc-msg').val() + '"';

					if (jQuery('#wptime').is(':checked'))
						output += ' wptime="true"';


					output += ']' + tinyMCEPopup.editor.selection.getContent() + '[/scheduled]';

					tinyMCEPopup.execCommand('mceInsertContent', 0, output);
					tinyMCEPopup.editor.execCommand('mceRepaint');
					tinyMCEPopup.editor.focus();
					// Return
					tinyMCEPopup.close();
				};
			</script>
			<style type="text/css">
				td.info {
					vertical-align: top;
					color: #777;
					width: 150px;
				}
			</style>

			<title><?php _e( "Scheduled Content", 'sc' ); ?></title>
		</head>
		<body style="display: none">
		<form onsubmit="insertSchedule();return false;" action="#">

			<div id="general_panel" class="panel current">
				<fieldset>
					<table border="0" cellpadding="4" cellspacing="0">
						<tr>
							<td><label for="sc-type"><?php _e( "Schedule", 'sc' ); ?></label></td>
							<td>
								<select id="sc-type" name="sc-type">
									<option value="monthly"><?php _e( "Monthly", 'sc' ); ?></option>
									<option value="weekly"><?php _e( "Weekly", 'sc' ); ?></option>
									<option value="onetime"><?php _e( "Onetime", 'sc' ); ?></option>
								</select>
							</td>
						</tr>
						<tr id="sc-monthly-row">
							<td><label for="sc-monthdate"><?php _e( "Date", 'sc' ); ?></label></td>
							<td>
								<select id="sc-monthdate" name="sc-monthdate">
									<?php
									for ( $i = 1; $i <= 31; $i ++ ) {
										echo '<option value="' . $i . '">' . $i . '</option>';
									}
									?>
								</select>
							</td>
							<td class="info"><?php _e( "Choose the day of each month to display the content.", 'sc' ); ?></td>
						</tr>
						<tr id="sc-weekly-row" style="display:none;">
							<td><label for="sc-weekdays-0"><?php _e( "Weekdays", 'sc' ); ?></label></td>
							<td>
								<label><input type="checkbox" class="sc-weekdays" name="sc-weekdays[]"
								              value="0"/><?php _e( "Sun.", 'sc' ); ?></label>
								<label><input type="checkbox" class="sc-weekdays" name="sc-weekdays[]"
								              value="1"/><?php _e( "Mon.", 'sc' ); ?></label>
								<label><input type="checkbox" class="sc-weekdays" name="sc-weekdays[]"
								              value="2"/><?php _e( "Tues.", 'sc' ); ?></label><br/>
								<label><input type="checkbox" class="sc-weekdays" name="sc-weekdays[]"
								              value="3"/><?php _e( "Wed.", 'sc' ); ?></label>
								<label><input type="checkbox" class="sc-weekdays" name="sc-weekdays[]"
								              value="4"/><?php _e( "Thurs.", 'sc' ); ?></label>
								<label><input type="checkbox" class="sc-weekdays" name="sc-weekdays[]"
								              value="5"/><?php _e( "Fri.", 'sc' ); ?></label><br/>
								<label><input type="checkbox" class="sc-weekdays" name="sc-weekdays[]"
								              value="6"/><?php _e( "Sat.", 'sc' ); ?></label>
							</td>
							<td class="info"><?php _e( "Choose what days of each week to display the content.", 'sc' ); ?></td>
						</tr>
						<tr id="sc-onetime-row" style="display:none;">
							<td><label for="sc-date"><?php _e( "Onetime Date", 'sc' ); ?></label></td>
							<td>
								<select id="sc-year" name="sc-year">
									<?php
									$now  = date( 'Y' );
									$till = $now + 10;
									for ( $i = $now; $i <= $till; $i ++ ) {
										echo '<option value="' . $i . '">' . $i . '</option>';
									}
									?>
								</select> -
								<select id="sc-month" name="sc-month">
									<?php
									for ( $i = 1; $i <= 12; $i ++ ) {
										$num = ( strlen( $i ) < 2 ) ? '0' . $i : $i;
										echo '<option value="' . $num . '">' . $num . '</option>';
									}
									?>
								</select> -
								<select id="sc-day" name="sc-day">
									<?php
									for ( $i = 1; $i <= 31; $i ++ ) {
										$num = ( strlen( $i ) < 2 ) ? '0' . $i : $i;
										echo '<option value="' . $num . '">' . $num . '</option>';
									}
									?>
								</select>
							</td>
							<td class="info"><?php _e( "Choose the onetime date to display the content.", 'sc' ); ?></td>
						</tr>
					</table>
				</fieldset>
				<br/>
				<fieldset>
					<table border="0" cellpadding="4" cellspacing="0">
						<tr>
							<td><label for="sc-hours"><?php _e( "Open Time", 'sc' ); ?></label></td>
							<td>
								<select id="sc-hours" name="sc-hours">
									<?php
									for ( $i = 1; $i <= 12; $i ++ ) {
										$num = ( strlen( $i ) < 2 ) ? '0' . $i : $i;
										echo '<option value="' . $num . '">' . $num . '</option>';
									}
									?>
								</select>:
								<select id="sc-minutes" name="sc-minutes">
									<?php
									for ( $i = 0; $i <= 59; $i ++ ) {
										$num = ( strlen( $i ) < 2 ) ? '0' . $i : $i;
										echo '<option value="' . $num . '">' . $num . '</option>';
									}
									?>
								</select>
								<select id="sc-ampm" name="sc-ampm">
									<option value="AM"><?php _e( "AM", 'sc' ); ?></option>
									<option value="PM"><?php _e( "PM", 'sc' ); ?></option>
								</select><br/>
								<label id="sc-utcoffset"><input type="checkbox" name="wptime"
								                                id="wptime"><?php _e( "Use WordPress Time Instead of GMT", 'sc' ); ?>
								</label>
							</td>
							<td class="info"><?php _e( "Choose the time you want to begin displaying the content.", 'sc' ); ?></td>
						</tr>
						<tr>
							<td><label for="sc-date"><?php _e( "Open Length", 'sc' ); ?></label></td>
							<td>
								<label><select id="sc-lyears" name="sc-lyears">
										<?php
										for ( $i = 0; $i <= 10; $i ++ ) {
											echo '<option value="' . $i . '">' . $i . '</option>';
										}
										?>
									</select> <?php _e( "Year(s)", 'sc' ); ?></label>
								<label><select id="sc-ldays" name="sc-ldays">
										<?php
										for ( $i = 0; $i <= 364; $i ++ ) {
											echo '<option value="' . $i . '">' . $i . '</option>';
										}
										?>
									</select> <?php _e( "Day(s)", 'sc' ); ?></label><br/>
								<label><select id="sc-lhours" name="sc-lhours">
										<?php
										for ( $i = 0; $i <= 23; $i ++ ) {
											$selected = ( $i == 1 ) ? ' selected="selected"' : '';
											echo '<option value="' . $i . '"' . $selected . '>' . $i . '</option>';
										}
										?>
									</select> <?php _e( "Hour(s)", 'sc' ); ?></label>
								<label><select id="sc-lmins" name="sc-lmins">
										<?php
										for ( $i = 0; $i <= 59; $i ++ ) {
											echo '<option value="' . $i . '">' . $i . '</option>';
										}
										?>
									</select> <?php _e( "Min", 'sc' ); ?></label>
							</td>
							<td class="info"><?php _e( "Choose how long to display the content.", 'sc' ); ?></td>
						</tr>
						<tr>
							<td><label for="sc-msg"><?php _e( "Message", 'sc' ); ?></label></td>
							<td>
								<label><input type="text" id="sc-msg" name="sc-msg" value="" style="width:100%"/></label>
							</td>
							<td
								class="info"><?php _e( "Optional - This message displays when the content is not available.", 'sc' ); ?></td>
						</tr>
					</table>
				</fieldset>
			</div>

			<div class="mceActionPanel">
				<div style="float: left">
					<input type="button" id="cancel" name="cancel" value="<?php esc_attr_e( "Cancel", 'sc' ); ?>"
					       onclick="tinyMCEPopup.close();"/>
				</div>

				<div style="float: right">
					<input type="submit" id="insert" name="insert" value="<?php esc_attr_e( "Insert", 'sc' ); ?>"/>
				</div>
			</div>
		</form>
		</body>
		</html>
		<?php
		exit( 0 );
	}

	/**
	 * @see    http://codex.wordpress.org/TinyMCE_Custom_Buttons
	 */
	function tinymce_register_button( $buttons ) {
		array_push( $buttons, "separator", "scheduled" );

		return $buttons;
	}

	/**
	 * @see    http://codex.wordpress.org/TinyMCE_Custom_Buttons
	 */
	function tinymce_add_plugin( $plugin_array ) {
		$plugin_array['scheduled'] = plugins_url(dirname(plugin_basename(__FILE__)) . '/includes/editor_plugin.js');

		return $plugin_array;
	}

} //end class

//load class
$_GLOBALS['ScheduledContent'] = new ScheduledContent();